### PR TITLE
only throw when the load_balancing_policy and specific fields of common_lb_config are set

### PR DESCRIPTION
**Pull request by wbpcode**
Signed-off-by: wbpcode <wangbaiping@corp.netease.com>
Commit Message: only throw when the load_balancing_policy and specific fields of common_lb_config are set
Additional Description:

An exception will be throwed when the common_lb_config and load_balancing_policy are set in the same time in the previous implemention.
But in fact, It should only throw an exception when the specific fields of common_lb_config and load_balancing_policy are set.
This PR correct it to make the behavior more appropriate.

Risk Level: Low.
Testing: Unit.
Docs Changes: n/a.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

File: source/common/upstream/upstream_impl.cc [hunk 1]
```diff
@@ -1187,22 +1187,25 @@ ClusterInfoImpl::ClusterInfoImpl(
 // Configures the load balancer based on config.load_balancing_policy
 void ClusterInfoImpl::configureLbPolicies(const envoy::config::cluster::v3::Cluster& config,
                                           Server::Configuration::ServerFactoryContext& context) {
-  if (config.has_lb_subset_config()) {
+  // Check if load_balancing_policy is set first.
+  if (!config.has_load_balancing_policy()) {
     throw EnvoyException(
-        fmt::format("cluster: LB policy {} cannot be combined with lb_subset_config",
+        fmt::format("cluster: load_balancing_policy requires field load_balancing_policy to be set",
```
[the remark this answers is not in the document]
Comment (wbpcode) [reply source/common/upstream/upstream_impl.cc:1193]: right.

**Review by adisuissa — COMMENTED** [review]
Thanks, overall LGTM, small nits.

File: source/common/upstream/upstream_impl.cc [hunk 2]
```diff
@@ -1223,7 +1224,8 @@ void ClusterInfoImpl::configureLbPolicies(const envoy::config::cluster::v3::Clus
 
   if (load_balancer_factory_ == nullptr) {
     throw EnvoyException(fmt::format(
-        "Didn't find a registered load balancer factory implementation for cluster: '{}'", name_));
+        "cluster: didn't find a registered load balancer factory implementation for cluster: '{}'",
```
[the remark this answers is not in the document]
Comment (wbpcode) [reply source/common/upstream/upstream_impl.cc:1227]: get it.

**Comment by wbpcode** [comment]
Friendly ping @adisuissa

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
/assign-from @envoyproxy/senior-maintainers

**Comment by wbpcode** [comment]
Seems that @lizan is too busy for this.

/assign-from @envoyproxy/senior-maintainers

**Review by lizan — APPROVED** [review]

**Pull request merged**